ddm-admin-client/build.rs

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2022 Oxide Computer Company
+
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use quote::quote;
+use serde::Deserialize;
+use std::collections::BTreeMap;
+use std::env;
+use std::fs;
+use std::path::Path;
+
+// TODO-cleanup This is a copy/paste of (a subset of) omicron_package::Config.
+// We ought to depend on it directly, but that creates a circular dependency
+// between crates:
+//
+// us -> omicron_package -> omicron_sled_agent -> us
+//
+// omicron_package only depends on sled-agent for zone management; maybe that
+// could be broken out into a separate crate?
+#[derive(Deserialize, Debug)]
+#[serde(tag = "type", rename_all = "lowercase")]
+pub enum ExternalPackageSource {
+    Prebuilt { repo: String, commit: String, sha256: String },
+    Manual,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ExternalPackage {
+    pub source: ExternalPackageSource,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Config {
+    #[serde(default, rename = "external_package")]
+    pub external_packages: BTreeMap<String, ExternalPackage>,
+}
+
+fn main() -> Result<()> {
+    // Find the current maghemite repo commit from our package manifest.
+    let manifest = fs::read("../package-manifest.toml")
+        .context("failed to read ../package-manifest.toml")?;
+    println!("cargo:rerun-if-changed=../package-manifest.toml");
```

Comment:
Nit, should we also rerun if `ddm-admin-client/src/lib.rs` changes?

Reply:
I don't think so - lib.rs consumes the output of this build script; I'm not sure what could change in it that would warrant rerunning the build script. But I could certainly be missing something - what do you think?

Reply:
Ah, that's right. Nevermind!